lib/cli.mjs

```diff
@@ -452,114 +424,87 @@ export default async function main(argv) {
+		options.ignoreDisables = ignoreDisables;
 	}
 
-	if (cli.flags.cacheLocation) {
-		optionsBase.cacheLocation = cli.flags.cacheLocation;
+	if (isBoolean(disableDefaultIgnores)) {
+		options.disableDefaultIgnores = disableDefaultIgnores;
 	}
 
-	if (cli.flags.cacheStrategy) {
-		optionsBase.cacheStrategy = cli.flags.cacheStrategy;
+	if (isBoolean(cache)) {
+		options.cache = cache;
 	}
 
-	if (cli.flags.fix) {
-		optionsBase.fix = cli.flags.fix;
+	if (isString(cacheLocation)) {
+		options.cacheLocation = cacheLocation;
 	}
 
-	if (cli.flags.outputFile) {
-		optionsBase.outputFile = cli.flags.outputFile;
+	if (isString(cacheStrategy)) {
+		options.cacheStrategy = cacheStrategy;
 	}
 
-	const reportNeedlessDisables = cli.flags.reportNeedlessDisables;
-	const reportInvalidScopeDisables = cli.flags.reportInvalidScopeDisables;
-	const reportDescriptionlessDisables = cli.flags.reportDescriptionlessDisables;
-
-	if (reportNeedlessDisables) {
-		optionsBase.reportNeedlessDisables = reportNeedlessDisables;
+	if (isBoolean(fix)) {
+		options.fix = fix;
 	}
 
-	if (reportInvalidScopeDisables) {
-		optionsBase.reportInvalidScopeDisables = reportInvalidScopeDisables;
+	if (isBoolean(reportNeedlessDisables)) {
+		options.reportNeedlessDisables = reportNeedlessDisables;
 	}
 
-	if (reportDescriptionlessDisables) {
-		optionsBase.reportDescriptionlessDisables = reportDescriptionlessDisables;
+	if (isBoolean(reportInvalidScopeDisables)) {
+		options.reportInvalidScopeDisables = reportInvalidScopeDisables;
 	}
 
-	const maxWarnings = cli.flags.maxWarnings;
-
-	if (maxWarnings !== undefined) {
-		optionsBase.maxWarnings = maxWarnings;
+	if (isBoolean(reportDescriptionlessDisables)) {
+		options.reportDescriptionlessDisables = reportDescriptionlessDisables;
 	}
 
-	if (cli.flags.help) {
-		cli.showHelp(EXIT_CODE_SUCCESS);
-
-		return;
+	if (isNumber(maxWarnings)) {
+		options.maxWarnings = maxWarnings;
 	}
 
-	if (cli.flags.version) {
-		cli.showVersion();
-
-		return;
+	if (isBoolean(allowEmptyInput)) {
+		options.allowEmptyInput = allowEmptyInput;
 	}
 
-	if (cli.flags.allowEmptyInput) {
-		optionsBase.allowEmptyInput = cli.flags.allowEmptyInput;
+	// Add input/code into options
+	if (cli.input.length > 0) {
+		options.files = cli.input;
+	} else {
+		options.code = await getStdin();
 	}
 
-	// Add input/code into options
-	/** @type {OptionBaseType} */
-	const options = cli.input.length
-		? {
-				...optionsBase,
-				files: cli.input,
-		  }
-		: await getStdin().then((stdin) => ({
-				...optionsBase,
-				code: stdin,
-		  }));
-
-	if (cli.flags.printConfig) {
-		return printConfig(options)
+	if (printConfigFlag) {
+		await printConfig(options)
 			.then((config) => {
 				process.stdout.write(JSON.stringify(config, null, '  '));
 			})
 			.catch(handleError);
+
+		return;
 	}
 
-	if (!options.files && !options.code && !cli.flags.stdin) {
-		cli.showHelp(EXIT_CODE_SUCCESS);
+	if (!options.files && !options.code && !stdin) {
+		showHelp();
```

Comment:
So the default is `===` to `EXIT_CODE_SUCCESS`?

Reply:
question

@ybiquitous Do we have a regression ?
If so do we need an additional test?
cf #5150

Reply:
I didn't change the behavior. `showHelp()` is a local helper and exits with `0`. See https://github.com/stylelint/stylelint/pull/7129/files#diff-d1470268bb8c3a42692b2c5249d7d31832ceeed1b1f16fbdbc814f8013c36b31R357